contracts/HubPool.sol

```diff
@@ -453,8 +454,8 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
         // Since _exchangeRateCurrent() reads this contract's balance and updates contract state using it, it must be
         // first before transferring any tokens to this contract to ensure synchronization.
         uint256 lpTokensToMint = (l1TokenAmount * 1e18) / _exchangeRateCurrent(l1Token);
-        ExpandedIERC20(pooledTokens[l1Token].lpToken).mint(msg.sender, lpTokensToMint);
```

Comment:
Moved token transfer "interaction" after any effects to follow CEI

Reply:
nice. doing this + having the guard is best posible. importantly, you are doing this _after_ the calls that modify tracking state so no risk in this change.